### PR TITLE
update to use Transform Stream API (for Serialport 6 compat)

### PR DESCRIPTION
**Pull request by Holger-Will**

### CODE REVIEW

**Comment by RafaelKa** [comment]
This PR will close #8 and #9

File: ESP3Parser.js [hunk 1]
```diff
@@ -2,160 +2,112 @@
...
-
-	var callbackForNextByte = waitForSyncByte;
-
-	this.tryToFillPackets = function(emitterU, buffer) {
-		emitter = emitterU;
-		for (var offset = 0; offset < buffer.length; offset++) {
-			callbackForNextByte(buffer[offset]);
+class EnoceanParser extends Transform {
+	constructor(options = {}) {
+		super(options)
```
Comment (RafaelKa) [inline ESP3Parser.js]: Suggestion 1 : Keep "Object Mode"  
```suggestion
		super(Object.assign(options, {readableObjectMode: true}));
```

File: ESP3Parser.js [hunk 2]
```diff
@@ -2,160 +2,112 @@
...
-	/**
-	 * Emits fetched ESP3 packet on data event.
-	 *
-	 */
-	function emitFetchedESP3Packet() {
-		var out = currentESP3Packet;
-		currentESP3Packet = new ESP3Packet();
-		emitter.emit("data", out);
+	emitFetchedESP3Packet() {
+		var out = this.currentESP3Packet.getRawBuffer();
```
Comment (RafaelKa) [inline ESP3Parser.js]: Suggestion 1 : Keep "Object Mode"  
```suggestion
		var out = this.currentESP3Packet;
```

File: test/esp3parser.js [hunk 3]
```diff
@@ -18,11 +19,11 @@ describe('serialport enocean parser', function(){
     describe('ESP3Packet.getRawBuffer()', function() {
         it('returns identical buffer', function() {
             for (var key in telegrams) {
-                var telegramm = new Buffer(telegrams[key], 'hex');
+                var telegramm = Buffer.from(telegrams[key], 'hex');
```
Comment (RafaelKa) [inline test/esp3parser.js]: Whole `for` block:
```
                const telegramm = Buffer.from(telegrams[key], 'hex');
                const spy = sinon.spy();
                const parser = new Esp3Parser({readableObjectMode: false});
                parser.on('data', spy);
                parser.write(telegramm);
                assert.equal(spy.getCall(0).args[0].getRawBuffer().toString('hex'), telegramm.toString('hex'));
```

File: ESP3Parser.js [hunk 4]
```diff
@@ -2,160 +2,112 @@
...
+		this.buffer = Buffer.alloc(0)
+		this.currentESP3Packet=new ESP3Packet()
+		this.tmp=null
+		this.callbackForNextByte = this.waitForSyncByte;
+	}
+	_transform(chunk, encoding, cb) {
+		this.tryToFillPackets(chunk)
+		cb()
+	}
+	tryToFillPackets(chunk) {
```
Comment (RafaelKa) [inline ESP3Parser.js]: Suggestion 2: 
would propose to move this function directly to `_transform()`

**Comment by Holger-Will** [comment]
i can not merge... i don't have write permission on your master branch... :-(

**Comment by RafaelKa** [comment]
> i can not merge... i don't have write permission on your master branch... :-(

accepted invite for collaboration?

**Pull request merged**